menas/src/main/scala/za/co/absa/enceladus/rest/services/DatasetService.scala

```diff
@@ -42,8 +44,8 @@ class DatasetService @Autowired()(datasetMongoRepository: DatasetMongoRepository
   }
 
   override def create(newDataset: Dataset, username: String): Future[Dataset] = {
-    val dataset = Dataset(
-      name = newDataset.name,
+      val dataset = Dataset(
+      name = validateEntityName(newDataset.name,"Dataset"),
```

Comment:
How about we keep the the validation call outside of the entity's definition? The validation function returns its input value or throws, we don't need to clutter the entity declaration with this.

Reply:
thanks for the explanation, made the changes